### PR TITLE
Publish privacy enhancements

### PR DESCRIPTION
**Pull request by Jesus89**
Closes https://github.com/CartoDB/cartoframes/issues/1286

### CODE REVIEW

File: examples/publish_and_share/publish_a_protected_visualization.ipynb [hunk 1]
```diff
@@ -4,9 +4,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Publish a private visualization\n",
+    "# Publish a protected visualization\n",
```
Comment (alrocar) [inline examples/publish_and_share/publish_a_protected_visualization.ipynb]:
```suggestion
    "# Publish a password protected visualization\n",
```

File: examples/publish_and_share/publish_a_protected_visualization.ipynb [hunk 2]
```diff
@@ -4,9 +4,9 @@
...
     "\n",
-    "This example illustrate how to publish a private visualization.\n",
+    "This example illustrate how to publish a protected visualization using a private dataset.\n",
```
Comment (alrocar) [inline examples/publish_and_share/publish_a_protected_visualization.ipynb]:
```suggestion
    "This example illustrate how to publish a password protected visualization using a private dataset.\n",
```

File: cartoframes/viz/map.py [hunk 3]
```diff
@@ -274,23 +274,23 @@ def get_content(self):
...
-    def publish(self, name, table_name=None, credentials=None, password=None):
+    def publish(self, name, password, table_name=None, credentials=None):
         """Publish the map visualization as a CARTO custom visualization (aka Kuviz).
 
         Args:
-            name (str): The Kuviz name on CARTO
+            name (str): The Kuviz name on CARTO.
+            password (str): By setting it, your Kuviz will be protected by
+                password. When someone tries to show the Kuviz, the password
+                will be requested. To disable password you must set it to None.
```
Comment (alrocar) [inline cartoframes/viz/map.py]: I haven't reviewed this before, but I'm unsure if referring to the term `Kuviz` makes sense. To me it looks like an internal term. We are not adding any reference to know more about what a Kuviz is neither, so maybe we could just use `visualization` or `map`. What do you think?

Comment (Jesus89) [reply cartoframes/viz/map.py]: Yes, visualization  :+1:

**Comment by alrocar** [comment]
Acceptance 🍏

**Pull request merged**